src/core/lib/event_engine/posix_engine/posix_engine.cc

```diff
@@ -911,57 +922,48 @@ PosixEventEngine::CreatePosixListener(
       "EventEngine::CreateListener is not supported on this platform");
 }
 
-#endif  // GRPC_PLATFORM_SUPPORTS_POSIX_POLLING
+#endif  // defined(GRPC_POSIX_SOCKET_TCP)
 
 #if GRPC_POSIX_SOCKET_TCP && GRPC_ENABLE_FORK_SUPPORT && \
     GRPC_POSIX_FORK_ALLOW_PTHREAD_ATFORK
 
 void PosixEventEngine::AfterFork(OnForkRole on_fork_role) {
-#if GRPC_PLATFORM_SUPPORTS_POSIX_POLLING
-  PosixEventPoller* poller = GetPollerChecked();
-#endif  // GRPC_PLATFORM_SUPPORTS_POSIX_POLLING
   if (on_fork_role == OnForkRole::kChild) {
-    if (grpc_core::IsEventEngineForkEnabled()) {
-      AfterForkInChild();
-    } else {
-#if GRPC_PLATFORM_SUPPORTS_POSIX_POLLING
-      poller->HandleForkInChild();
-#endif  // GRPC_PLATFORM_SUPPORTS_POSIX_POLLING
-    }
+    AfterForkInChild();
+  }
```

Comment:
Why is the check on IsEventEngineForkEnabled() removed ? 

should there be an `else if (poller_ != nullptr) { poller_->HandleForkInChild(); }` here if the IsEventEngineForkEnabled() returned false ?